packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/webserver/functions/functions_rpc_interface.py

```diff
@@ -0,0 +1,325 @@
+        WEBSERVER_RPC_NAMESPACE,
+        TypeAdapter(RPCMethodName).validate_python("register_function"),
+        function=function,
+    )
+    return TypeAdapter(RegisteredFunction).validate_python(
+        result
+    )  # Validates the result as a RegisteredFunction
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def get_function(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    function_id: FunctionID,
+) -> RegisteredFunction:
+    result = await rabbitmq_rpc_client.request(
+        WEBSERVER_RPC_NAMESPACE,
+        TypeAdapter(RPCMethodName).validate_python("get_function"),
+        function_id=function_id,
+    )
+    return TypeAdapter(RegisteredFunction).validate_python(result)
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def get_function_input_schema(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    function_id: FunctionID,
+) -> FunctionInputSchema:
+    result = await rabbitmq_rpc_client.request(
+        WEBSERVER_RPC_NAMESPACE,
+        TypeAdapter(RPCMethodName).validate_python("get_function_input_schema"),
+        function_id=function_id,
+    )
+    return TypeAdapter(FunctionInputSchema).validate_python(result)
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def get_function_output_schema(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    function_id: FunctionID,
+) -> FunctionOutputSchema:
+    result = await rabbitmq_rpc_client.request(
+        WEBSERVER_RPC_NAMESPACE,
+        TypeAdapter(RPCMethodName).validate_python("get_function_output_schema"),
+        function_id=function_id,
+    )
+    return TypeAdapter(FunctionOutputSchema).validate_python(result)
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def delete_function(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    function_id: FunctionID,
+) -> None:
+    result = await rabbitmq_rpc_client.request(
+        WEBSERVER_RPC_NAMESPACE,
+        TypeAdapter(RPCMethodName).validate_python("delete_function"),
+        function_id=function_id,
+    )
+    assert result is None  # nosec
+    return result
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def list_functions(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    pagination_offset: int,
+    pagination_limit: int,
+) -> tuple[list[RegisteredFunction], PageMetaInfoLimitOffset]:
+    result: tuple[list[RegisteredFunction], PageMetaInfoLimitOffset] = (
+        await rabbitmq_rpc_client.request(
+            WEBSERVER_RPC_NAMESPACE,
+            TypeAdapter(RPCMethodName).validate_python("list_functions"),
+            pagination_offset=pagination_offset,
+            pagination_limit=pagination_limit,
+        )
+    )
+    assert isinstance(result, tuple)
+    TypeAdapter(list[RegisteredFunction]).validate_python(
+        result[0]
+    )  # Validates the result as a list of RegisteredFunctions
+    assert isinstance(result[1], PageMetaInfoLimitOffset)  # nosec
+    return result
```

Comment:
Q: can you try and see if this works
```suggestion
    return TypeAdapter(tuple[list[RegisteredFunction], PageMetaInfoLimitOffset]).validate_python(result)
```

Reply:
done